### ISSUE TITLE
unnecessary-constructor prompts to remove necessary constructor

### ISSUE THREAD

**Issue by MaceWindu** [palantir/tslint#4905]
### Bug Report

- __TSLint version__: 6.0.0-beta1
- __TypeScript version__: 3.6.4
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
class A {
        private _x: number;
        protected constructor() {
            this._x = 43;
        }
    }

    class B extends A {
       // unnecessary-constructor  Remove unnecessary empty constructor.
        public constructor() {
            super();
        }
    }
```

with `tslint.json` configuration:

```json
"unnecessary-constructor": {
            "options": {
                "check-super-calls": true
            }
        }
```

#### Actual behavior

`unnecessary-constructor  Remove unnecessary empty constructor.` error reported

#### Expected behavior

rule should take into account visibility of base constructor

**Comment by JoshuaKGoldberg** [comment]
Indeed. As a followup to #4511 this should not be reported.

Accepting PRs for.. the next ~10 days! Per #4534. 😄

**Comment by MaceWindu** [comment]
Whoops, didn't found that issue. Anyway, probably I need to find some time and finally move to eslint :smile:

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
